### PR TITLE
Feature/add repositories params to overall tir and tip

### PR DESCRIPTION
**Pull request by victorpablosceruelo**
In this PR we modify code to ensure that docGen receives the parameters it needs to generate documents TIP and overall TIR. 
Basically we send the repositories info we have in docGen.
It is important to note that we retrieve the info we send just before calling docGen, so we keep the previous behavior.

### CODE REVIEW

**Review by s2oBCN — CHANGES_REQUESTED** [review]

File: src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy [hunk 1]
```diff
@@ -10,7 +11,19 @@ class LeVADocumentParamsMapper {
...
+        }
+
+        if (documentType.isOverallTIR() || documentType.isTIP()) {
+            return build([ repositories: buildRepositoriesData() ])
+        }
+
+        return build()
+    }
+
+    Map build(Map data = [:]) {
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy:26]: changed

File: src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy [hunk 2]
```diff
@@ -20,6 +33,36 @@ class LeVADocumentParamsMapper {
         ] << data
     }
 
+
+    protected List buildRepositoriesData() {
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy:37]: changed

File: src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy [hunk 3]
```diff
@@ -20,6 +33,36 @@ class LeVADocumentParamsMapper {
...
+        List result = []
+        for (Map repo: project.repositories) {
+            result << [ buildRepoData(repo) ]
+        }
+        return result
+    }
+
+    protected Map buildRepoData(Map repo) {
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy:45]: changed

File: src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy [hunk 4]
```diff
@@ -20,6 +33,36 @@ class LeVADocumentParamsMapper {
...
+                "git": [
+                    "branch": gitMap.branch,
+                    "commit": gitMap.commit,
+                    "baseTag": gitMap.baseTag,
+                    "targetTag": gitMap.targetTag,
+                ]
+            ]
+        ]
+
+        // Is null until finalize stage.
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy:60]: changed

File: src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy [hunk 5]
```diff
@@ -20,6 +33,36 @@ class LeVADocumentParamsMapper {
         ] << data
     }
 
+
+    protected List buildRepositoriesData() {
+        List result = []
+        for (Map repo: project.repositories) {
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply src/org/ods/orchestration/mapper/LeVADocumentParamsMapper.groovy:39]: changed to noncps

File: test/groovy/org/ods/core/test/usecase/levadoc/fixture/ProjectRepositoryFixture.groovy [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                references: "myReferences-A"
+            ]
+        ]
+        return repo
+    }
+
+    // a wrapper closure around executing a string
+    // can take either a string or a list of strings (for arguments with spaces)
+    // prints all output, complains and halts on error
+    String runCommand(String cmd) {
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply test/groovy/org/ods/core/test/usecase/levadoc/fixture/ProjectRepositoryFixture.groovy:112]: removed

File: test/groovy/org/ods/core/test/usecase/levadoc/fixture/ProjectRepositoryFixture.groovy [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        } else {
+            bitbucketUrl = bitbucketUrl.replace(HTTP_HEAD, "")
+        }
+
+        bitbucketUrl = (isHttps ? HTTPS_HEAD : HTTP_HEAD) +
+            "${bitbucketUsername}:${bitbucketPassword}@${bitbucketUrl}"
+        return bitbucketUrl
+    }
+
+    private def loadDefaultRepoMetadata(repo) {
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply test/groovy/org/ods/core/test/usecase/levadoc/fixture/ProjectRepositoryFixture.groovy:95]: removed

File: test/groovy/org/ods/core/test/usecase/levadoc/utils/LeVADocumentParamsMapperWithLogging.groovy [hunk 8]
```diff
@@ -26,5 +27,21 @@ class LeVADocumentParamsMapperWithLogging extends LeVADocumentParamsMapper {
...
+    Map buildWithRepositories() {
+        Map result
+        try {
+            result = super.buildWithRepositories()
+        } catch(Throwable e) {
+            log.error(e.getMessage(), e)
+            throw e
+        }
+
+        log.info("--- INPUT GENERATED BY LeVADocumentParamsMapper.buildWithRepositories(data) (start) ---")
```
[the remark this answers is not in the document]
Comment (victorpablosceruelo) [reply test/groovy/org/ods/core/test/usecase/levadoc/utils/LeVADocumentParamsMapperWithLogging.groovy:40]: changed

**Pull request closed without merging**